src/stories/development/engineering_rules.stories.mdx

```diff
@@ -2,23 +2,107 @@ import { Meta } from '@storybook/addon-docs'
 
 <Meta title="Development/Engineering Rules" />
 
-# Engineering rules (!!!WIP!!!)
+# Engineering rules
 
-#### How do we write our custom components?
+🚧 This is where we note down our agreements and rules as a developers team. Sor far, this is
+rather a notepad. And its content is very much subject to change.
 
-- We use CSS Modules.
-- We do not use Sass/SCSS for the Design System
-  - because we don’t need it for variables and nesting anymore. PostCSS allows us to do nesting.
-  - because it will reduce bundle sizes.
-  - because we want to have so many dependencies.
-- We don't use a CSS Framework, we build our own
+## Foundation
 
-### CSS class splitting
+### CSS Framework
 
-CSS classes must always be written in one piece, as well in Slim as in JS.
+**We don’t rely on any existing CSS framework, we build our own.** After having worked with
+and evaluated a couple of well-established CSS/UI frameworks (like Bootstrap, Foundation,
+Tailwind, Bulma, Semantic, Chakra) we have learned and decided that our best bet is to build
+our design system, and accordingly, our CSS system completely from scratch.
 
-- bad: `label-${success ? 'success' : 'error'}`
-- good: `success ? 'label-success' : 'label-error'`
-- bad: [https://css-tricks.com/the-sass-ampersand/#article-header-id-10](https://css-tricks.com/the-sass-ampersand/#article-header-id-10)
+Our main reasons for this decision:
 
-This helps to find all occurrences of classes while developing and, more importantly, [PurgeCSS](https://purgecss.com/) can detect used classes and purge the correct classes.
+1. We won’t depend on framework maintainers and their decisions.
+2. We will be flexible enough to meet any specific need of a large platform like ours.
+3. **Because we can. And we ❤️ CSS.** We are confident enough to write our stylesheets on our own.
+  CSS is _not_ some obscure, secret science that needs to be handled somewhere under the hood
+  of a 3rd-party framework. It’s a great technology that we’ve grown up with over the last
+  two decades. It’s a language we love.
+
+### CSS methodologies
+
+Not using a 3rd-party framework does not mean completely re-inventing the wheel. We are eager to
+learn what the community knows and to build upon great ideas that others brought up.
+
+#### ITCSS
+…
+
+#### BEM
+…
+
+## Tooling
+
+### CSS preprocessor
+- We use [PostCSS](https://postcss.org/).
+- We do no longer need any other (traditional) preprocessor like Sass or LESS, because:
+  - PostCSS allows for the same things we used to achieve with Sass/LESS:
+    - variables,
+    - nesting,
+    - mix-ins, functions …
+  - Relying only on PostCSS spares us some dependencies.
+  - It will reduce bundle sizes (?)
+
+### CSS-in-JS
+- For component-specific styles we employ CSS-in-JS, namely [CSS Modules](https://github.com/css-modules/css-modules).
```

Comment:
Very good and non-trivial question. That’s why it’s the last one for me to answer 😅 …

To be honest, I was a bit ignorant counting CSS Modules as CSS-in-JS. It’s just using one technology (an imperative programming language, namely JavaScript) to handle another one (declarative stylesheet language, namely CSS), isn’t it? While, traditionally, we aimed [to separate those two aspects](https://www.w3.org/wiki/The_web_standards_model_-_HTML_CSS_and_JavaScript) of web documents ([style](https://www.w3.org/wiki/The_web_standards_model_-_HTML_CSS_and_JavaScript#CSS_.E2.80.94_let.E2.80.99s_add_some_style) and [behaviour](https://www.w3.org/wiki/The_web_standards_model_-_HTML_CSS_and_JavaScript#JavaScript_.E2.80.94_adding_behaviour_to_web_pages)) we now mingle it all together, to some degree, at least. Instead of loading CSS files directly (in the document) we load _CSS in JavaScript_. Hence, CSS Modules, just like any other approach that makes CSS dependent on JavaScript, for me fell under the category “CSS-in-JS”.

That being said, and now that you asked, I learned that [people distinguish between CSS Modules and CSS-in-JS](https://dev.to/alexsergey/css-modules-vs-css-in-js-who-wins-3n25). Turns out, “CSS-in-JS” is commonly used for techniques where stylesheets are literally included in the JavaScript code, rather than imported from an external CSS file. Considering that either way incorporates loading stylesheet data into some JavaScript object to process it and apply it to the DOM (be it via generated and linked CSS files or via CSSOM) to me this is pretty samey. But for the sake of common understanding **I shall no longer mix it up**!

Reply:
```suggestion
- We manage component-specific styles within the component code scope using [CSS Modules](https://github.com/css-modules/css-modules).
```

Reply:
Could you also change the headline (to CSS Modules or something..)?